rubin_scheduler/scheduler/schedulers/queue_manager.py

```diff
@@ -0,0 +1,133 @@
+__all__ = ("BaseQueueManager",)
+
+import healpy as hp
+import numpy as np
+
+from rubin_scheduler.scheduler.utils import IntRounded
+
+
+class BaseQueueManager:
+    """Class for managing a queue of desired observations.
+
+    NOTE: If using masking basis functions, one should
+    ensure Survey objects are using the same masks, else
+    one will end up in a loop where surveys propose visits
+    that the QueueManager always rejects.
+
+    Parameters
+    ----------
+    detailers : `list`
+        List of detailers to apply to observations as they are
+        selected from the queue. Mostly for setting camera
+        rotation angles dynamically, or skipping observations
+        that have become clouded out or gotten close to zenith.
+    basis_functions : `list`
+        List of basis functions. Assumed to be only used for
+        masking, thus no weights for the basis functions.
+
+    """
+
+    def __init__(self, detailers=None, basis_functions=None):
+        if detailers is None:
+            self.detailers = []
+        else:
+            self.detailers = detailers
+        if basis_functions is None:
+            self.basis_functions = []
+        else:
+            self.basis_functions = basis_functions
+        self.desired_observations_array = None
+        # Array to track which desired_observations_array
+        # have been observed
+        self.need_observing = False
+
+    def flush_queue(self):
+        self.desired_observations_array = None
+        self.need_observing = False
+
+    def set_queue(self, observation_array):
+        self.desired_observations_array = observation_array
+        self.need_observing = np.ones(observation_array.size, dtype=bool)
+
+    def add_observation(self, observation):
+        if self.desired_observations_array is not None:
+            match_indx = np.where(self.desired_observations_array["target_id"] == observation["target_id"])
+            self.need_observing[match_indx] = False
+
+    def add_observations_array(self, observation_array):
+        if self.desired_observations_array is not None:
+            indx = np.isin(self.desired_observations_array["target_id"], observation_array["target_id"])
+            self.need_observing[indx] = False
+
+    def compute_reward(self, conditions):
+        reward = 0
+        for bf in self.basis_functions:
+            reward += bf(conditions)
+        return reward
+
+    def find_valid_indx(self, conditions):
```

Comment:
Would a name like "find_valid_desired_observations" be more helpful in understanding what this does?

Reply:
I like putting `observations` in the name